docs/proposal/2022-11-22-botkube-plugins-integration.md

```diff
@@ -0,0 +1,498 @@
+## Integrate plugins into Botkube
+
+Created on 2022-11-22 by Mateusz Szostok ([@mszostok](https://github.com/mszostok))
+
+| Status     |
+|------------|
+| `PROPOSED` |
+
+<!-- TOC -->
+  * [New syntax](#new-syntax)
+  * [Use cases](#use-cases)
+    * [Defining list of plugin repositories and enabling a given plugin](#defining-list-of-plugin-repositories-and-enabling-a-given-plugin)
+    * [Defining executors aliases](#defining-executors-aliases)
+    * [Validating plugin configuration](#validating-plugin-configuration)
+      * [Plugin configuration](#plugin-configuration)
+    * [Passing configuration to a given plugin](#passing-configuration-to-a-given-plugin)
+    * [Refreshing already loaded plugins](#refreshing-already-loaded-plugins)
+    * [Supporting multi OS and architectures](#supporting-multi-os-and-architectures)
+    * [Releasing Botkube plugins](#releasing-botkube-plugins)
+  * [Implementation details](#implementation-details)
+    * [E2E testing](#e2e-testing)
+    * [How plugins are stored](#how-plugins-are-stored)
+    * [Do we want to decouple the interfaces between plugin Go implementation and gRPC?](#do-we-want-to-decouple-the-interfaces-between-plugin-go-implementation-and-grpc)
+    * [Do we want to have a separate go mod for each executor/source?](#do-we-want-to-have-a-separate-go-mod-for-each-executorsource)
+    * [Botkube directory structure](#botkube-directory-structure)
+  * [Alternatives](#alternatives)
+    * [The `plugins` object syntax](#the-plugins-object-syntax)
+    * [How a given plugins is configured](#how-a-given-plugins-is-configured)
+<!-- TOC -->
+
+## Motivation
+
+This proposal describe how the [plugin system](2022-09-28-botkube-plugin-system.md) can be integrated with the Botkube core. It describes both the configuration syntax changes and expected behavior.
+
+## New syntax
+
+This section describes the necessary changes in the syntax. **It's backward compatible.**
+
+1. A new `plugins` property. It allows to specify a list of repository from where the plugins can be downloaded.
+   ```yaml
+   plugins:
+     cacheDir: "/tmp"
+     repositories:
+       botkube: https://plugins.botkube.io/botkube.yaml
+       huseyinbabal: https://raw.githubusercontent.com/huseyinbabal/botkube-plugins/main/index.json
+   ```
+
+2. The `executors` definition now can refer to the executor plugins specified in a given repository.
+   ```yaml
+   executors:
+     'plugin-based':
```

Comment:
I didn't get this field, it is for back ground compatibility to not corrupt our confs under executors?